### PR TITLE
Developer Onboarding documentation updates

### PR DESCRIPTION
**Pull request by relandboyle**
Updates to Developer Onboarding document:
Add .vscode folder and workspace settings
Update section about Firebase service account / private key
Update section about Mongo connection string to include database name

Fixes # N/A

### CODE REVIEW

**Review by tevansbuilds — APPROVED** [review]
All updates are interpretable and assist in spinning up dev, as well as making the doc more developer friendly.

File: .github/docs/DEVELOPER_ONBOARDING.md [hunk 1]
```diff
@@ -53,25 +54,22 @@ Each developer should create their own local .env files. The project-level .giti
 
 There should be one .env file in the **client/** folder and one .env file in the **server/** folder. These files live at the root of the client and server applications, respectively.
 
-```
-Note: There is no .env file at the project root level!
```
Comment (colinvandergraaf) [inline .github/docs/DEVELOPER_ONBOARDING.md:57]: FYI:  I came across this md syntax for different types of notes to add visual cues to markdown files.  Might be extra here, but good to know about. 
```
> [!TIP]
> Optional information to help a user be more successful.

> [!IMPORTANT]  
> Crucial information necessary for users to succeed.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

> [!CAUTION]
> Negative potential consequences of an action.


>[!DANGER]
> Dangerous action that could cause irreversible damage.
```

**Review by colinvandergraaf — APPROVED** [review]
LGTM!

File: .github/docs/DEVELOPER_ONBOARDING.md (hunk 1 above)
Comment (relandboyle) [reply .github/docs/DEVELOPER_ONBOARDING.md:57]: @colinvandergraaf I will use those annotations on the next update.

**Pull request merged**